runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,428 @@
+                      'description': '(Optional) WAF id to associate with the '
+                                     'distribution.'},
+    }
+
+    def create_template(self):
+        # type: () -> None
+        """Create the Blueprinted template for Auth@Edge."""
+        self.template.set_version('2010-09-09')
+        self.template.set_description(
+            'Authorization@Edge Static Website - Bucket, Lambdas, and Distribution'
+        )
+
+        self.add_parameters()
+
+        # Resources
+        bucket = self.add_bucket()
+        self.add_bucket_policy(bucket)
+        oai = self.add_origin_access_identity()
+        allow_access = self.allow_cloudfront_access_on_bucket(bucket, oai)
+        # @TODO: Make this available in Auth@Edge
+        lambda_function_associations = []
+
+        if self.directory_index_specified:
+            rewrite_role = self.add_index_rewrite_role()
+            index_rewrite = self.add_cloudfront_directory_index_rewrite(rewrite_role)
+            index_rewrite_version = self.add_cloudfront_directory_index_rewrite_version(
+                index_rewrite
+            )
+            lambda_function_associations = self.get_directory_index_lambda_association(
+                lambda_function_associations,
+                index_rewrite_version
+            )
+
+        # Auth@Edge Lambdas
+        lambda_execution_role = self.add_lambda_execution_role()
+        check_auth_lambda = self.add_lambda(
+            'CheckAuth',
+            'Check Authorization information for request',
+            'check_auth',
+            lambda_execution_role
+        )
+        http_headers_lambda = self.add_lambda(
+            'HttpHeaders',
+            'Additional Headers added to every response',
+            'http_headers',
+            lambda_execution_role
+        )
+        parse_auth_lambda = self.add_lambda(
+            'ParseAuth',
+            'Parse the Authorization Headers/Cookies for the request',
+            'parse_auth',
+            lambda_execution_role
+        )
+        refresh_auth_lambda = self.add_lambda(
+            'RefreshAuth',
+            'Refresh the Authorization information when expired',
+            'refresh_auth',
+            lambda_execution_role
+        )
+        sign_out_lambda = self.add_lambda(
+            'SignOut',
+            'Sign the User out of the application',
+            'sign_out',
+            lambda_execution_role
+        )
+
+        # CloudFront Distribution
+        distribution_options = self.get_distribution_options(
+            bucket,
+            oai,
+            lambda_function_associations,
+            check_auth_lambda['version'],
+            http_headers_lambda['version'],
+            parse_auth_lambda['version'],
+            refresh_auth_lambda['version'],
+            sign_out_lambda['version']
+        )
+        distribution = self.add_cloudfront_distribution( # noqa pylint: disable=unused-variable
```

Comment:
This should use the `add_cloudfront_distribution` method from the base blueprint to ensure the shared resource between them (the CF distribution) always has the same logical identifier.

Reply:
Sorry I'm not following here. The cloudfront distribution is vastly different between the two blueprints, if I swap this in to use the one from the base then this won't function correctly. What am I missing?